quarto/src/Repository/GameRepository.php

```diff
@@ -7,67 +7,100 @@
 
-  private $em;
-
-  public function __construct(EntityManagerInterface $em) {
-    $this->em = $em;
-  }
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+    }
 
-  public function findGameById(string $id) {
-    return $this->em->find('App:Game', $id);
-  }
+    public function findGameById(string $id)
+    {
+        return $this->em->find('App:Game', $id);
+    }
 
-  public function getOpenedGamesList(array $tokenList) {
-    if (count($tokenList) === 0) {
-      $tokenList=["NOTHING"];
+    public function getOpenedGamesList(array $tokenList)
+    {
+        if (count($tokenList) === 0) {
+            $tokenList=["NOTHING"];
+        }
+        $query = $this->em->createQuery("SELECT g.id_game idGame,
+            g.number_players numberPlayers,
+            g.solo_game soloGame,
+            g.closed,
+            '' token
+        FROM App:Game g  " .
+        "WHERE g.closed=false " .
+        "AND g.number_players = 1 " .
+        "AND g.token_player_one NOT IN (:player_one_token) " .
+        "AND g.token_player_two NOT IN (:player_two_token) " .
+        "ORDER BY g.id_game ASC");
+        $query->setParameter('player_one_token', $tokenList);
+        $query->setParameter('player_two_token', $tokenList);
+        return $query->getResult();
     }
-    $query = $this->em->createQuery("SELECT g FROM App:Game g  " .
-    "WHERE g.closed=false " .
-    "AND g.number_players = 1 " .
-    "AND g.token_player_one NOT IN (:player_one_token) " .
-    "AND g.token_player_two NOT IN (:player_two_token) " .
-    "ORDER BY g.id_game ASC");
-    $query->setParameter('player_one_token', $tokenList);
-    $query->setParameter('player_two_token', $tokenList);
-    return $query->getResult();
-  }
 
-  public function getCurrentGamesList(array $tokenList) {
-    $query = $this->em->createQuery("SELECT g FROM App:Game g  " .
-    "WHERE g.closed=false  " .
-    "AND (g.token_player_one IN (:player_one_token)  " .
-    "  OR g.token_player_two IN (:player_two_token))  " .
-    "ORDER BY g.id_game ASC");
-    $query->setParameter('player_one_token', $tokenList);
-    $query->setParameter('player_two_token', $tokenList);
-    return $query->getResult();
-  }
+    public function getCurrentGamesList(array $tokenList)
+    {
+        $queryOne = $this->em->createQuery("SELECT g.id_game idGame,
+            g.number_players numberPlayers,
+            g.solo_game soloGame,
+            g.closed,
+            g.token_player_one token
+        FROM App:Game g  " .
+        "WHERE g.closed=false  " .
+        "AND g.token_player_one IN (:player_one_token) " .
+        "ORDER BY g.id_game ASC");
+        $queryOne->setParameter('player_one_token', $tokenList);
+        $arrayOne = $queryOne->getResult();
 
-  public function getOnlySpectateGamesList(array $tokenList) {
-    if (count($tokenList) === 0) {
-      $tokenList=["NOTHING"];
+        $queryTwo = $this->em->createQuery("SELECT g.id_game idGame,
```

Comment:
You can't do only one request ?

Reply:
I thought but, I searched online how to make "UNION" in PHP with doctrine, all responses said to make 2 resquest and join them in PHP.
So ....